server/tools.js

```diff
@@ -247,6 +248,17 @@ module.exports = function () {
         }
     };
 
+    this.storage = multer.diskStorage({
+        destination: function(req, file, callback) {
+            callback(null, String(req.body.upload_path)) 
```

Comment:
Is it possible that upload_path is null ? If it is we have to make sure nothing will break here...